database/gormdb_tasks.go

```diff
@@ -98,3 +99,45 @@ func (db *GormDB) SynchronizeAssignmentTasks(course *pb.Course, taskMap map[uint
 
 	return createdTasks, updatedTasks, err
 }
+
+// CreatePullRequest creates a pull request
+func (db *GormDB) CreatePullRequest(pullRequest *pb.PullRequest) error {
+	if !pullRequest.Validate() {
+		return errors.New("pull request is not valid for creation")
+	}
+	pullRequest.SetDraft()
+	return db.conn.Create(pullRequest).Error
+}
+
+// GetPullRequest returns the pull request matching the given query
+func (db *GormDB) GetPullRequest(query *pb.PullRequest) (*pb.PullRequest, error) {
+	var pullRequest pb.PullRequest
+	if err := db.conn.Where(query).Last(&pullRequest).Error; err != nil {
+		return nil, err
+	}
+	return &pullRequest, nil
+}
+
+// HandleMergingPR handles merging a pull request
+func (db *GormDB) HandleMergingPR(pullRequest *pb.PullRequest) error {
+	// If a pull request has not been approved, it should not have been merged.
```

Comment:
Move comment to details to the doc comment, since this is not explaining specifics of the code, but the general idea.